assets/src/design-system/components/modal/index.js

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useContext } from 'react';
+import ReactModal from 'react-modal';
+import PropTypes from 'prop-types';
+import { createGlobalStyle, ThemeContext } from 'styled-components';
+import { rgba } from 'polished';
+
+const ADMIN_TOOLBAR_HEIGHT = 32;
+const ADMIN_MENU_WIDTH = 160;
+const ADMIN_MENU_FOLDED_WIDTH = 36;
+const CONTENT_CLASS = 'WebStories_ReactModal__Content';
+const OVERLAY_CLASS = 'WebStories_ReactModal__Overlay';
+
+export const GlobalStyle = createGlobalStyle`
+  .${OVERLAY_CLASS} {
+    opacity: 0;
+    transition: opacity 0.1s ease-out;
+    top: ${ADMIN_TOOLBAR_HEIGHT}px !important;
+    left: ${ADMIN_MENU_WIDTH}px !important;
+  }
+
+  body.folded .${OVERLAY_CLASS} {
+    left: ${ADMIN_MENU_FOLDED_WIDTH}px !important;
+  }
+
+  .${OVERLAY_CLASS}.ReactModal__Overlay--after-open {
+    opacity: 1;
+  }
+
+  .${OVERLAY_CLASS}.ReactModal__Overlay--before-close {
+    opacity: 0;
+  }
+`;
+
+const customStyles = {
+  overlay: (theme) => ({
+    position: 'fixed',
+    top: 0,
+    left: 0,
+    right: 0,
+    bottom: 0,
+    zIndex: 10,
+    backgroundColor: rgba(theme?.colors.bg.workspace || '#1B1D1C', 0.96),
```

Comment:
Need to ask Sam what background color she wants here, this is just copied over from existing modal overlay and i subbed out the color for new theme